src/error_ext.rs

```diff
@@ -0,0 +1,107 @@
+use Fail;
+use std::error::Error;
+use std::fmt::{self, Debug, Display};
+use std::sync::Mutex;
+
+/// Wrapper for std errors to make them Sync.
+///
+/// This exists to coerce existing types that are only `Error + Send + 'static`
+/// into a `Fail`-compatible representation.
+///
+/// Unfortunately, this requires wrapping the error in a `Mutex`, which must be
+/// locked for every `Debug`/`Display`.
+pub struct SyncErr<T>(Mutex<T>);
+
+impl<E> SyncErr<E> {
+    fn new(err: E) -> Self {
+        SyncErr(Mutex::new(err))
+    }
+}
+
+impl<T> Display for SyncErr<T>
+where
+    T: Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.lock().unwrap().fmt(f)
+    }
+}
+
+impl<T> Debug for SyncErr<T>
+where
+    T: Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.lock().unwrap().fmt(f)
+    }
+}
+
+impl<E: Display + Debug + Send + 'static> Fail for SyncErr<E> {}
+
+/// Extensions for std errors
+pub trait ErrorExt: Sized {
```

Comment:
What is the reason to have this extension trait? Shouldn't we just make `SyncFailure::new` public?

Reply:
I didn't really have a good reason tbh and was thinking about removing it. Publicizing `SyncFailure::new` works for me!

Reply:
It would make sense if it were possible to make it part of `Fail`, but since that isn't possible I don't think its worth adding a new trait just to get method syntax. :)

Reply:
True, and since it'll most often be a simple `.map_err(SyncFailure::new)` call, the method syntax doesn't even buy us much in the way of ergonomics.